apps/gcd/views/details.py

```diff
@@ -75,6 +91,94 @@ def show_publisher(request, publisher, preview=False):
     
     return paginate_response(request, publisher_series,
                              'gcd/details/publisher.html', vars)
+
+
+def publisher_monthly_covers(request,
+                             publisher_id,
+                             year=None,
+                             month=None,
+                             use_on_sale=True):
+    """
+    Display the covers for the monthly publications of a publisher.
+    """
+    publisher = get_gcd_object(Publisher, publisher_id)
+
+    if use_on_sale:
+        date_type = 'publisher_monthly_covers_on_sale'
+    else:
+        date_type = 'publisher_monthly_covers_pub_date'
+
+    # parts are generic and should be re-factored if we ever need it elsewhere
+    try:
+        if 'month' in request.GET:
+            year = int(request.GET['year'])
+            month = int(request.GET['month'])
+            # do a redirect, otherwise pagination links point to current month
+            return HttpResponseRedirect(
+              urlresolvers.reverse(date_type,
+                                   kwargs={'publisher_id': publisher_id,
+                                           'year': year,
+                                           'month': month} ))
+        if year:
+            year = int(year)
+        if month:
+            month = int(month)
+    except ValueError:
+        year = None
```

Comment:
What's the rationale for ignoring a malformed year instead of raising an error?  I'm not saying I disagree, just not quite following.